### DISCUSSION TITLE
GitHub paged visibility for private repo

### DISCUSSION THREAD

**Discussion by chenx2018** [community/community#22374] [Other Feature Feedback, Questions, & Ideas]
<p>I am a GitHub Pro user. I believe I can use GitHub Pages for my private repo, but the <code>select visibility</code> button not show. How to change visibility to only available to selected users instead of public?</p>
<p>According to <a href="https://docs.github.com/en/pages/getting-started-with-github-pages/changing-the-visibility-of-your-github-pages-site" rel="noopener nofollow ugc">This tutorial </a>, there should be bottom to change visibility</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/c/d/cd30e31e00af8b4b2a70f4fa98d2481b7555c68e.jpeg" data-download-href="/uploads/short-url/thcCxWNOKJ0aSn2RTESHJoA4D8y.jpeg?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/37342983/181070627-5069cff0-0cba-41a9-ba37-37c796abb3ae.jpeg" alt="image" data-base62-sha1="thcCxWNOKJ0aSn2RTESHJoA4D8y" width="581" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/c/d/cd30e31e00af8b4b2a70f4fa98d2481b7555c68e_2_581x500.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/c/d/cd30e31e00af8b4b2a70f4fa98d2481b7555c68e_2_871x750.jpeg 1.5x, https://aws1.discourse-cdn.com/github/original/3X/c/d/cd30e31e00af8b4b2a70f4fa98d2481b7555c68e.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/c/d/cd30e31e00af8b4b2a70f4fa98d2481b7555c68e_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1125×967 100 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by jfullstackdev** [comment]
<p>where is the screenshot of the one you are referring too?</p>
<p>that <code>Source</code> is referring to whether you’ll be hosting your repo in <code>main</code>, <code>gh-pages</code> or <code>docs</code>,</p>

**Comment by chenx2018** [comment]
<p>There should be <code>select visibility</code> Button above or below <code>Source</code> Section.</p>

**Comment by jfullstackdev** [comment]
<p>did you already deploy that? because GitHub Pages is still not enabled, that’s what I’m seeing,</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/b/b/bb6ef40617b3f0def2dda3bdba72819fde07a366.png" data-download-href="/uploads/short-url/qK70c6322xZiXN2V8WIgWOo9miW.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/47092464/181070628-5cb9e29b-9a99-4867-b5ca-7f2b6005cf14.png" alt="image" data-base62-sha1="qK70c6322xZiXN2V8WIgWOo9miW" width="642" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/b/b/bb6ef40617b3f0def2dda3bdba72819fde07a366_2_642x500.png, https://aws1.discourse-cdn.com/github/original/3X/b/b/bb6ef40617b3f0def2dda3bdba72819fde07a366.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/b/b/bb6ef40617b3f0def2dda3bdba72819fde07a366.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/b/b/bb6ef40617b3f0def2dda3bdba72819fde07a366_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">911×709 119 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>the other one might be your subscription, I’m not very sure of your subscription, but it says here,</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/b/ab95d8a63247df5a6c193e097cbbcc65f2b25d44.png" data-download-href="/uploads/short-url/otUGSDHxVAyUV1YiNY2WE3WzkZ6.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/47092464/181070630-a1c02993-b96d-40fe-b782-8261ecc25130.png" alt="image" data-base62-sha1="otUGSDHxVAyUV1YiNY2WE3WzkZ6" width="690" height="156" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/b/ab95d8a63247df5a6c193e097cbbcc65f2b25d44_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">903×205 16.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by chenx2018** [comment]
<p>After I deployed, the my GitHub pages becomes public and I can’t find select visibility button in Setting-pages section. I’m a GitHub Pro user.</p>

**Comment by jfullstackdev** [comment]
<p>okay so we are now sure, your private repo does not have GitHub Enterprise Cloud, right?</p>

**Comment by chenx2018** [comment]
<p>I don’t have GitHub Enterprise Cloud.</p>

**Comment by jfullstackdev** [comment]
<p>that’s it,</p>
<p>you can have the Enterprise trial period if you want, they are offering that too,</p>

**Comment by chenx2018** [comment]
<p>GitHub pro does not provide this function? Ok</p>

**Comment by byrneh** [comment, marked as answer]
<p><a class="mention" href="/chenx2018">@chenx2018</a><br>
Reference: <a href="https://docs.github.com/en/get-started/learning-about-github/githubs-products" rel="noopener nofollow ugc">githubs-products</a><br>
Only mentions this feature use for GitHub Enterprise license use</p>
